internal/controller/component/controller.go

```
@@ -910,3 +919,57 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Named("component").
 		Complete(r)
 }
+
+// ─── Module Delegation ───────────────────────────────────────────────────────
+
+// handleModuleDelegation checks whether the community module referenced by the
+// ComponentType is installed (by probing for its sentinel CRD) and sets the
+// appropriate status condition. When installed, the core controller does nothing
+// further — the module's own controller handles reconciliation.
```

golangci-lint (2.12.1)

[error] File is not properly formatted

(gci)